DOCS-8624-Scripting-examples-edits-kt (#929)
mulesoft/docs-connectors: scripting/2.0/modules/ROOT/pages/edits/scripting-module-examples.adoc

= Scripting Module - Examples - Mule 4
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> Scripting Module examples help you use binding variables and access the registry.

Reviewer: ```suggestion
The following Scripting Module examples show you how to use binding variables and how to access the registry.
```

== Use Binding Variables Example

The following Scripting Module XML example shows how to:

* Use a flow variable in the scripting code, such as `vars.increment`
* Use scripting parameters by referencing them, such as `initialValue`
* Log some information by using the `log` variable, such as `log.info("....")`

> * Use the `result` variable to set the result of the execution.

Reviewer: ```suggestion
* Use the `result` variable to set the result of the execution
```

* Define the payload in `vars.increment + Number(payload) + initialValue`

[source,xml,linenums]
----

<set-variable variableName="increment" value="#[22]" />
<scripting:execute engine="ECMAScript">
    <scripting:code>
        log.info("Incrementing payload by " + (vars.increment + initialValue))
        result = vars.increment + Number(payload) + initialValue
    </scripting:code>
    <scripting:parameters><![CDATA[#[{ initialValue: 10 }]]]></scripting:parameters>
</scripting:execute>
----

== Access the Registry Example
The following Scripting Module XML example shows how to:

* Access the registry

> * Make a flow stop or start

Reviewer: ```suggestion
* Stop or start a flow
```
